docs/docs/topics/webhooks.mdx

```diff
@@ -0,0 +1,177 @@
+---
+title: Webhooks
+---
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import CodeBlock from '@theme/CodeBlock';
+import FastapiWebhookReciever from '!!raw-loader!../media/fastapi_webhook_reciever.txt';
+
+# Webhooks
+
+The **Webhook System** enables external notifications based on the [Events](./infrahub-event.mdx) system within InfraHub, which also powers the [Activity Log](./activity-log.mdx).
+
+## Configuring webhooks
+
+Infrahub support two different types of webhooks:
+
+- `Standard Webhook` sends event information to an external endpoint.
+- `Custom Webhook` allows the ability to tie events to a [transformation](../topics/transformation.mdx) to configure the payload being sent to an external endpoint.
+
+:::note
+
+Custom webhooks might be useful in situations where the endpoint needs a specific payload to be successful. A use-case that might need this would be triggering external actions in platforms like GitHub and Gitlab.
+
+:::
+
+Webhooks can be configured to trigger on specific event types, such as:
+
+- `infrahub.branch.merged`
+- `infrahub.node.updated`
+
+This allows you to tailor webhooks for specific use cases.
+
+### Configuring branch types
+
+Webhooks can also be configured to trigger based on branch types:
+
+- **Other Branches** – All branches except the default branch.
+- **Default Branch** – Only the default branch.
+- **All Branches** – Every branch.
+
+This flexibility ensures that webhooks only trigger when relevant, depending on the use case. For example, you might configure a webhook to notify an external system only when a user makes changes directly to the default branch.
+
+### Webhook configuration details
+
+- **Description**: A description can be set to identify the webhooks purpose.
+- **URL**: The destination URL can be configured, such as `http://ansible-eda:8080`.
+
+### Shared key for security
+
+A **shared key** can be configured to sign webhook requests. This key ensures the **authenticity** and **integrity** of the webhook message.
+
+When a webhook is sent:
+
+1. The sender generates a signature using the shared key.
+2. The signature is included in the request headers.
+
+The receiver then uses the same shared key to verify the signature, confirming that the message has not been tampered with and that it is from a trusted source.
+
+<details>
+  <summary>Python FastAPI Example</summary>
+  <CodeBlock language="python" title="webhook-reciever.py">{FastapiWebhookReciever}</CodeBlock>
+
+</details>
+<!-- vale off -->
+### Validating HTTPS certificates
+<!-- vale on -->
+
+When using webhooks over HTTPS, you can configure your system to validate certificates to ensure that the receiver has a valid HTTPS certificate, adding an extra layer of security.
+
+## Example payloads
+
+Here are some example payloads of some events that may be sent.
+<!-- vale off -->
+<Tabs>
+  <TabItem value="infrahub.node.created" label="infrahub.node.created" default>
+```json
+{
```

Comment:
FYI, the format of the payload will change once #5956 has been merged

Reply:
I have changed the payload examples now.